### PR TITLE
0.1.7

### PR DESCRIPTION
**Pull request by maxwolff**
quick bugfixes

* getBuildFile searched for `contract.json` instead of `contracts.json`, breaking `eventBuilder` in money market
* getContract didnt actually set provider, breaking goerli repl

### CODE REVIEW

File: src/contract.ts [hunk 1]
```diff
@@ -16,7 +15,7 @@ interface ContractBuild {
 }
 
 export function getBuildFile(build_dir: string, trace?: boolean): string {
-  const fileName = trace ? `contract-trace.json` : `contract.json`;
+  const fileName = trace ? `contract-trace.json` : `contracts.json`;
```
Comment (coburncoburn) [inline src/contract.ts]: decent chance this breaks open oracle or w/e other project this was being developed with, assuming contract.json works in at least one situation?

File: src/contract.ts [hunk 2]
```diff
@@ -69,8 +64,7 @@ export async function getContractAt(web3: Web3, name: string, build_dir: string,
 export async function deployContract(web3: Web3, network: string, name: string, args: any[], build_dir: string, trace: boolean, defaultOptions: SendOptions, sendOptions: SendOptions): Promise<{contract: Contract, receipt: TransactionReceipt}> {
   const contractBuild = await getContractBuild(name, build_dir, trace);
   const contractAbi = JSON.parse(contractBuild.abi);
```
Comment (coburncoburn) [inline src/contract.ts]: I don't think you need the contractAbi her anymore since getContract doing that work now

File: src/contract.ts (hunk 1 above)
Comment (maxwolff) [reply src/contract.ts]: investigating.. though looks like open oracle also uses `contracts.json`

Comment (coburncoburn) [reply src/contract.ts]: guess just a typo all along

Comment (maxwolff) [reply src/contract.ts]: actually, this typo just got introduced in [0.1.6](https://github.com/compound-finance/saddle/commit/da9118d0ccfa785e4e4b83022542d7d89089057d#diff-bb179d2c3824efa379684e18610b93a3)

Comment (maxwolff) [reply src/contract.ts]: actually, it looks like open-oracle does use `contract.json` in the docker builds. cant really decipher why (or how). @hayesgm ?

Comment (hayesgm) [reply src/contract.ts]: Yeah, we were awkwardly assuming the build_dir was `./build` in places, even though it's actually specified in the settings. I fixed that in 0.1.6, but I guess I fixed it to be `contract.json` not `contracts.json`. We can make this fix, but we should also name this `contracts-trace.json` too, then, right?

File: src/contract.ts [hunk 3]
```diff
@@ -54,10 +52,7 @@ export async function getContractABI(name: string, build_dir: string, trace: boo
 export async function getContract(web3: Web3, name: string, build_dir: string, trace: boolean, defaultOptions: SendOptions): Promise<Contract> {
   const contractBuild = await getContractBuild(name, build_dir, trace);
   const contractAbi = JSON.parse(contractBuild.abi);
-  (<any>newContract).setProvider((<any>web3).currentProvider);
-  const contract = <Contract>(new (<any>newContract)(contractAbi));
-
-  return contract;
+  return new web3.eth.Contract(contractAbi, undefined, defaultOptions);;
```
Comment (hayesgm) [inline src/contract.ts:55]: Hm, any idea why this was done the old way (minus the type issues)?

Comment (coburncoburn) [reply src/contract.ts:55]: I have a feeling that it felt better to use the subdependancy directly and avoid web3 magic, but unfortunately the name setProvider for the subdependancy is non obvious that you need to also pass in the unlocked accounts https://github.com/ethereum/web3.js/blob/1.x/packages/web3-eth-contract/src/index.js#L301

**Review by jflatow — APPROVED** [review]

**Pull request merged**